atc/db/team_test.go

```diff
@@ -2161,6 +2161,262 @@ var _ = Describe("Team", func() {
+				Expect(otherUpdatedResource.ID()).To(Equal(otherResource.ID()))
+			})
+
+			It("should handle when old resource has the same name as new resource", func() {
+				pipeline, _, err := team.SavePipeline(pipelineName, config, 0, false)
+				Expect(err).ToNot(HaveOccurred())
+
+				resource, _, _ := pipeline.Resource("some-resource")
+
+				config.Resources[0].Name = "some-resource"
+				config.Resources[0].OldName = "some-resource"
+
+				updatedPipeline, _, err := team.SavePipeline(pipelineName, config, pipeline.ConfigVersion(), false)
+				Expect(err).ToNot(HaveOccurred())
+
+				updatedResource, _, _ := updatedPipeline.Resource("some-resource")
+				Expect(updatedResource.ID()).To(Equal(resource.ID()))
+			})
+
+			It("should return an error when there is a swap with resource name", func() {
+				pipeline, _, err := team.SavePipeline(pipelineName, config, 0, false)
+				Expect(err).ToNot(HaveOccurred())
+
+				config.Resources[0].Name = "new-resource"
+				config.Resources[0].OldName = "some-resource"
+
+				config.Resources[1].Name = "some-resource"
+				config.Resources[1].OldName = "new-resource"
+
+				_, _, err = team.SavePipeline(pipelineName, config, pipeline.ConfigVersion(), false)
+				Expect(err).To(HaveOccurred())
+			})
+
+			Context("when new resource exists but is disabled", func() {
+				var pipeline db.Pipeline
+				var resource db.Resource
+				var rcv db.ResourceConfigVersion
+				var err error
+				var found bool
+
+				BeforeEach(func() {
+					pipeline, _, err = team.SavePipeline(pipelineName, config, 0, false)
+					Expect(err).ToNot(HaveOccurred())
+
+					resource, found, err = pipeline.Resource("some-resource")
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+
+					setupTx, err := dbConn.Begin()
+					Expect(err).ToNot(HaveOccurred())
+
+					brt := db.BaseResourceType{
+						Name: "some-type",
+					}
+
+					_, err = brt.FindOrCreate(setupTx, false)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(setupTx.Commit()).To(Succeed())
+
+					resourceScope, err := resource.SetResourceConfig(atc.Source{"some": "other-repository"}, atc.VersionedResourceTypes{})
+					Expect(err).NotTo(HaveOccurred())
+
+					err = resourceScope.SaveVersions(nil, []atc.Version{
+						{"disabled": "version"},
+					})
+					Expect(err).ToNot(HaveOccurred())
+
+					rcv, found, err = resourceScope.FindVersion(atc.Version{"disabled": "version"})
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+
+					err = resource.DisableVersion(rcv.ID())
+					Expect(err).ToNot(HaveOccurred())
+
+					versions, _, found, err := resource.Versions(db.Page{Limit: 3}, nil)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+					Expect(versions).To(HaveLen(1))
+					Expect(versions[0].Version).To(Equal(atc.Version{"disabled": "version"}))
+					Expect(versions[0].Enabled).To(BeFalse())
+				})
+
+				It("should successfully update resource name", func() {
```

Comment:
Maybe this should test that the version is still disabled after the rename, rather than checking that the IDs match up?